crates/common/validator/src/beacon_api_client/mod.rs

```diff
@@ -479,6 +485,64 @@ impl BeaconApiClient {
             })
         }
     }
+
+    pub async fn publish_block(
+        &self,
+        broadcast_validation: BroadcastValidation,
+        signed_beacon_block: SignedBeaconBlock,
+    ) -> anyhow::Result<(), ValidatorError> {
+        let response = self
+            .http_client
+            .execute(
+                self.http_client
+                    .post("/eth/v2/beacon/blocks".to_string())?
+                    .query(&[(
+                        "broadcast_validation",
+                        serde_json::to_string(&broadcast_validation)?,
+                    )])
+                    .header(ETH_CONSENSUS_VERSION_HEADER, VERSION)
+                    .body(signed_beacon_block.as_ssz_bytes())
+                    .build()?,
+            )
+            .await?;
+
+        if !response.status().is_success() {
+            return Err(ValidatorError::RequestFailed {
+                status_code: response.status(),
+            });
+        }
+
+        Ok(())
+    }
+
+    pub async fn publish_blinded_block(
+        &self,
+        broadcast_validation: BroadcastValidation,
+        signed_blinded_beacon_block: SignedBlindedBeaconBlock,
+    ) -> anyhow::Result<(), ValidatorError> {
+        let response = self
+            .http_client
+            .execute(
+                self.http_client
+                    .post("/eth/v2/beacon/blinded_blocks".to_string())?
+                    .query(&[(
+                        "broadcast_validation",
+                        serde_json::to_string(&broadcast_validation)?,
+                    )])
+                    .header(ETH_CONSENSUS_VERSION_HEADER, VERSION)
+                    .body(signed_blinded_beacon_block.as_ssz_bytes())
+                    .build()?,
```

Comment:
The `Content-Type` header must be specified right? or is that done automatically?

Reply:
This discussion shouldn't be marked as resolved, `The http client is initialized with the content header for ssz` this is set for the get case, I don't think the post case is handled properly here